### PR TITLE
Add ability to create a new Location from within the volunteers/new form

### LINKED ISSUE #334: Add ability to create a new Location from within the volunteers/new form
Users can select from existing Universities on `/volunteers/new` form.
We'd like a select2 that allows a create (if not found using typeahead) on Location
Location needs a `location_type`, so this should be defaulted to the string `university`

### PR DESCRIPTION
**Pull request by shivamvinsol**
Resolves #334 
### Description
Adds ability to create new location using `CreatableSelect`.

### Type of change
- New feature (non-breaking change which adds functionality)

### How Has This Been Tested?
Tested it on dev env.

### Screenshots
Screenshot 2019-10-24 at 7 31 28 PM

### CODE REVIEW

File: app/controllers/locations_controller.rb [hunk 1]
```diff
@@ -33,7 +33,13 @@ def create
     respond_to do |format|
       if @location.save
         format.html { redirect_to @location, notice: 'Location was successfully created.' }
-        format.json { render :show, status: :created, location: @location }
+        format.json do
+          if params[:source] == 'volunteers-form'
```
Comment (shivamvinsol) [inline app/controllers/locations_controller.rb:37]: I am not familiar with how the other json request worked, so added a source param to return json, instead of rendering show template.

File: app/javascript/src/components/volunteer-application-form/VolunteerApplicationForm.js [hunk 2]
```diff
@@ -62,6 +63,37 @@ class VolunteerApplicationForm extends React.Component {
     this.setState({ selectedUniversity, volunteerApplication: { ...this.state.volunteerApplication, university_location_id: selectedUniversity.value} })
   }
 
+  handleUniversityCreate(university) {
```
Comment (shivamvinsol) [inline app/javascript/src/components/volunteer-application-form/VolunteerApplicationForm.js:69]: Tried to maintain consistency with the existing fetch request.

**Comment by shivamvinsol** [comment]
@maebeale This PR is ready for review. Please let me know your thoughts

**Comment by maebeale** [comment]
Hello @shivamvinsol ! Very excited about this PR!

I'm getting an error when running this code (I don't have any existing Universities in my database, so that could be the edge case that needs to be handled). Would you be willing to explore it?

Also, are you able/willing to add tests for this feature?

**Comment by maebeale** [comment]
Screen Shot 2019-10-22 at 5 35 18 PM

**Comment by shivamvinsol** [comment]
@maebeale, thank you for review. I did check the code, and it seems the error is coming after you select "ok to call?", and has no connection with this PR. I can raise another PR to fix that.
Regarding tests, I can write rspecs for Rails code, but I am not familiar with react test cases.
But lets do that in another PR too. Hope thats fine with you?

**Comment by maebeale** [comment]
@shivamvinsol sounds great! my apologies for the delay to approve -- i hadn't seen your response above
